go/vt/vttablet/onlineddl/executor.go

```diff
@@ -838,30 +838,64 @@ func (e *Executor) killTableLockHoldersAndAccessors(ctx context.Context, tableNa
 		}
 	}
 	capableOf := mysql.ServerVersionCapableOf(conn.ServerVersion)
-	capable, err := capableOf(capabilities.PerformanceSchemaDataLocksTableCapability)
-	if err != nil {
-		return err
-	}
-	if capable {
-		{
-			// Kill connections that have open transactions locking the table. These potentially (probably?) are not
-			// actively running a query on our table. They're doing other things while holding locks on our table.
-			query, err := sqlparser.ParseAndBind(sqlProcessWithLocksOnTable, sqltypes.StringBindVariable(tableName))
-			if err != nil {
-				return err
-			}
-			rs, err := conn.Conn.ExecuteFetch(query, -1, true)
-			if err != nil {
-				return vterrors.Wrapf(err, "finding transactions locking table")
+	{
```

Comment:
Maybe extract out  into a common function and pass in the capability and query for the two cases?

Reply:
Agreed. They seem identical other than the capability and query so we could pass both into a function.

Reply:
done.